packages/flow-core/src/components/f-file-upload/f-file-upload.ts

```diff
@@ -0,0 +1,536 @@
+				if (this.type === "multiple") {
+					filesArr.forEach(item => {
+						this.selectedFiles?.push(item);
+					});
+				} else {
+					if (filesArr?.length === 1) {
+						this.selectedFiles = filesArr;
+					}
+				}
+				this.dispatchOnInput(e);
+			}
+		}
+	}
+
+	/**
+	 * select file(s) from browse options
+	 * @param e InputEvent
+	 */
+	selectFile(e: InputEvent) {
+		const fileInput = this.fileInputRef.value as HTMLInputElement;
+		const files = fileInput.files;
+		const filesArr = files ? Array.from(files) : [];
+		this.sizeLimitFlag =
+			this.maxSize && this.bytes ? filesArr.every(item => item.size < this.bytes) : true;
+		if (this.sizeLimitFlag) {
+			if (this.type === "multiple") {
+				filesArr.forEach(item => {
+					this.selectedFiles?.push(item);
+				});
+			} else {
+				this.selectedFiles = filesArr;
+			}
+			this.dispatchOnInput(e);
+		}
+	}
+
+	/**
+	 * dispatch input event
+	 * @param e Event
+	 */
+	dispatchOnInput(e: Event) {
+		e.stopPropagation();
+		/**
+		 * @event input
+		 */
+		const event = new CustomEvent("input", {
+			detail: {
+				value:
+					this.type === "single"
+						? this.selectedFiles
+							? this.selectedFiles[0]
+							: []
+						: this.selectedFiles
+			},
+			bubbles: true,
+			composed: true
+		});
+		this.value =
+			this.type === "single"
+				? this.selectedFiles
+					? this.selectedFiles[0]
+					: []
+				: this.selectedFiles;
+		this.dispatchEvent(event);
+		this.requestUpdate();
+	}
+
+	/**
+	 * on click remove the selected single file
+	 * @param e MouseEvent
+	 */
+	handleRemoveFile(e: MouseEvent) {
+		e.stopPropagation();
+		this.acceptedFilesFlag = true;
+		this.sizeLimitFlag = true;
+		this.selectedFiles = [];
+		this.dispatchOnInput(e);
+	}
+
+	/**
+	 * removes the clicked respective file from selection array
+	 * @param e MouseEvent
+	 * @param file contains clicked file
+	 */
+	handleRemoveRespectiveFile(e: MouseEvent, file: File) {
+		if (this.selectedFiles && this.selectedFiles?.length > 0) {
+			this.selectedFiles = this.selectedFiles.filter(item => item.name !== file.name);
+			this.dispatchOnInput(e);
+		}
+	}
+
+	/**
+	 * check if ellipsis happens on text and insert tooltip
+	 */
+	checkOverflowing() {
+		if (this.textOverflow) {
+			if (this.textOverflow.offsetWidth < this.textOverflow.scrollWidth) {
+				this.textOverflow.tooltip = this.textOverflow.innerHTML;
+			} else {
+				this.textOverflow.tooltip = "";
+			}
+		}
+	}
+
+	/**
+	 * in multiple file selection hovering on the particular file name may open the tooltip -if ellipsis is present
+	 * @param e MouseEvent
+	 */
+	handleMouseEnter(e: MouseEvent) {
+		const element = e.target as HTMLElement;
+		if (element?.offsetWidth < element?.scrollWidth) {
+			(e.target as FText).tooltip = (e.target as FText).innerText;
+		} else {
+			(e.target as FText).tooltip = "";
+		}
+	}
+
+	/**
+	 * has label slot
+	 */
+	_onLabelSlotChange() {
+		this._hasLabel = this._labelNodes.length > 0;
+	}
+
+	/**
```

Comment:
don't require slot change methods,

just do `@slotchange=${this.requestUpdate}`